packages/react-components/src/components/SendBox.tsx

```diff
@@ -233,12 +233,20 @@ export const SendBox = (props: SendBoxProps): JSX.Element => {
     }
 
     // we dont want to send empty messages including spaces, newlines, tabs
-    if (!EMPTY_MESSAGE_REGEX.test(textValue)) {
-      onSendMessage && onSendMessage(textValue);
+    // Message can be empty if there is a valid file upload
+    if (!EMPTY_MESSAGE_REGEX.test(textValue) || hasFile()) {
+      // Chat SDK doesn't allow sending messages with NULL content. For sending files without text message,
+      // we need to send a message with an empty string.
+      if (EMPTY_MESSAGE_REGEX.test(textValue)) {
```

Comment:
I don't think we need if else from 240-242, textValue will be always '' if you don't put anything there
Or we can do
onSendMessage && onSendMessage(textValue ?? '');

Reply:
We want to avoid sending spaces (as per my understanding).
If we remove this if else check, someone would be able to send `'               '` which will make the message appear strange.
This logic ensures that if someone has entered a bunch of spaces, we only send blank message

Reply:
These conditionals have become a bit convoluted.

Helper method would make the flow easier to follow, I think.

```
function sendSanitizedText(sendFunction: (sanitizedText: string) => void, textValue: string, hasFile) {
  if (!EMPTY_MESSAGE_REGEX.test(textValue)) {
    sendFunction(textValue);
    return;
  }
  if (hasFile) {
    // Send an empty message if there is a valid file to send.
    sendFunction('');
  }
}

...

// and here in the onClick callback
onSendMessage && sendSanitizedText(onSendMessage, textValue, hasFile());
```

Reply:
Looks more complicated to me now. :D